src/main.rs

```diff
@@ -1394,14 +1403,85 @@ async fn list_deadline(config: Config, args: &ListDeadline) -> Result<String, Er
 
 // // --- CONFIG ---
 
-async fn config_check_version(_args: &ConfigCheckVersion) -> Result<String, Error> {
-    match cargo::compare_versions(None).await {
-        Ok(Version::Latest) => Ok(format!("Tod is up to date with version: {VERSION}")),
-        Ok(Version::Dated(version)) => Err(Error::new(
-            "cargo",
-            &format!("Tod is out of date with version: {VERSION}, latest is:{version}"),
-        )),
-        Err(e) => Err(e),
+async fn config_check_version(
+    args: &ConfigCheckVersion,
+    mock_url: Option<String>,
+) -> Result<String, Error> {
```

Comment:
Because args.force and args.repo are used in a few different places, you can pattern match them out at the beginning of the function.

```rust
let ConfigCheckVersion{args, force} = args;
```

Not only does this clean up the args a bit, but now if we add another argument, the compiler will warn us that it needs to be added here and then warn us that it is unused.